app/code/Magento/GraphqlPlayground/etc/frontend/routes.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:App/etc/routes.xsd">
+    <router id="standard">
+        <route id="graphiql" frontName="graphiql">
```

Comment:
We must not introduce a new URL. The idea was to render GraphiQL, when a user opens `/graphql` route in the browser instead of GraphiQL or similar client. With the custom endpoint the value of this feature is pretty low, because developers will have to know about this alternative URL, and will still not getting any help when trying to open `graphql` in the browser.

Please investigate how it is possible to avoid new endpoint.

Reply:
can we use **user-agent** header to differentiate, I can think of that as the only possibility. Can you suggest if there is a better approach, or any native magento utility to achieve this.

Reply:
> can we use user-agent header to differentiate, I can think of that as the only possibility.

Would prefer we avoid doing anything with user agent strings - notoriously unreliable.

Can we use the following requirements for a request:

- Is a `GET`
- Does not have a `query` querystring value

If both are true, render the playground

Reply:
@DrewML Thanks for the suggestions. This is more clear than user-agent.